components/src/core/Drawer/DrawerRailItem.tsx

```diff
@@ -0,0 +1,259 @@
+import React, { useCallback } from 'react';
+import { Avatar, ButtonBase, Divider, Tooltip, Typography } from '@material-ui/core';
+import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
+import { PXBlueDrawerInheritableProperties, RAIL_WIDTH, RAIL_WIDTH_CONDENSED } from './Drawer';
+import color from 'color';
+import clsx from 'clsx';
+
+export type DrawerRailItemClasses = {
+    root?: string;
+    active?: string;
+    condensed?: string;
+    divider?: string;
+    icon?: string;
+    statusStripe?: string;
+    title?: string;
+    titleActive?: string;
+    ripple?: string;
+};
+
+export type DrawerRailItemProps = Pick<
+    PXBlueDrawerInheritableProperties,
+    | 'activeItemBackgroundColor'
+    | 'activeItemFontColor'
+    | 'activeItemIconColor'
+    | 'divider'
+    | 'itemFontColor'
+    | 'itemIconColor'
+    | 'onItemSelect'
+    | 'ripple'
+> & {
+    activeItem?: string;
+
+    // toggles the condensed style
+    condensed?: boolean;
+
+    // sets whether to hide the nav item
+    hidden?: boolean;
+
+    // icon on the left
+    icon?: JSX.Element;
```

Comment:
`icon` should be mandatory in a rail

Reply:
And should we make title optional for rails?

Reply:
I can make the type mandatory, but it won't have a real appreciable effect since users don't use the DrawerRailItem directly...all of the props passed to it are controlled by the DrawerNavItem, which currently uses NavItem as the type for items (that's a whole different can of worms...i tried to do NavItem | RailItem....but then all of the differences between the two started coming up everywhere...red squigglies like you wouldn't believe).

I think for the Rail, if they don't want a title, they can pass in `""` or use the condensed option.